cocos/network/WebSocket.h

```diff
@@ -238,7 +239,7 @@ class CC_DLL WebSocket
 
     struct lws*         _wsInstance;
     struct lws_context* _wsContext;
-    std::shared_ptr<bool> _isDestroyed;
+    std::shared_ptr<std::atomic<bool>> _isDestroyed;
```

Comment:
I think we couldn't do that.
Compiler will give you some errors:

![screen shot 2016-04-07 at 14 57 40](https://cloud.githubusercontent.com/assets/493372/14343078/1c9116e0-fcd1-11e5-8e9a-c75034c6752f.png)


, or capture the whole scope variable's reference by `[&]`